actgen/train.py

```diff
@@ -94,8 +95,25 @@ def setup(self):
             self.agent = RandomAgent(env.observation_space, env.action_space)
         elif self.params['agent'] == 'dqn':
             self.agent = DQNAgent(env.observation_space, env.action_space, self.params)
+        elif self.params['agent'] == 'action_dqn':
+            self.agent = ActionDQNAgent(env.observation_space, env.action_space, self.params)
         self.best_score = -np.inf
 
+        if self.params['regularization'] == 'l1':
+            regularizer = self.params['regularization'] + "_" + str(self.params['regularization_weight_l1'])
+        elif self.params['regularization'] == 'l2':
+            regularizer = self.params['regularization'] + "_" + str(self.params['regularization_weight_l2'])
+        elif self.params['regularization'] == 'dropout':
+            regularizer = self.params['regularization'] + "_" + str(self.params['dropout_rate'])
+        else:
+            regularizer = 'none'
+        self.file_name = self.params['agent'] + '_' \
+                    + 'seed' + str(self.params['seed']) + '_' \
+                    + regularizer
+        self.dir = self.params['results_dir'] + self.params['tag'] + '/'
+
+        utils.save_hyperparams(self.dir+self.file_name+'_hyperparams.csv', self.params)
```

Comment:
I added this line to make sure we always archive the hyperparameters we ended up using. Sometimes we'll want to override the defaults on the command line, and it's good to have a record of what the changes were.